weblate/trans/models/change.py

```diff
@@ -592,6 +592,9 @@ def auto_status(self):
     def get_action_display(self):
         if self.action in self.PLURAL_ACTIONS:
             return self.PLURAL_ACTIONS[self.action] % self.plural_count
+        if self.action == self.ACTION_NEW_SOURCE:
+            return self.get_source()
```

Comment:
Display will be messed up with plural strings (that's also reason for failing tests). The value should be rendered using `format_translation` template tag in `weblate/templates/last-changes-content.html`.

Also, this is wrong place to display content, this method returns action name and is rendered in the left column while browsing changes.